packages/data-stores/src/onboard/index.ts

```diff
@@ -40,7 +40,6 @@ export function register(): typeof STORE_KEY {
 			'goals',
 			'hasUsedDomainsStep',
 			'hasUsedPlansStep',
-			'hideFreePlan',
```

Comment:
also revert, or keep with intention (confirm)

Reply:
this lingered from some experimental code that got reverted. however, it's also one of the changes I want to explore next

For some reason, we persist `hideFreePlan` in the store, presumably to deal with redirections or something. However, this being persisted means the Onboard store will have a value for this when other flows/places attempt to query. One change I'd like to explore next is to remove all the props for hiding individual plans (most of them can already be skipped with "intent" here) e.g. `hideFreePlan` can be derived from store directly when a flow renders the plans-grid calypso wrapper (i.e. `PlansFeaturesMain` can just query the store for it). One less prop to pass around as it's a value already instantiated centrally in the store, and one less pattern (`hideFooPlan`) is a good idea. 

^ probably not if it's persisted though as it may get ugly

Reply:
I will leave this change to see if it leads to any breakage that we haven't spotted. We've been testing a lot of things with it, so hopefully this can safely not persist. It may be false hope though as I don't think there's actual use of it right now (last I recall, it gets overridden in the plans wrapper).

Reply:
Agree. Let's leave it as it is for now. I believe [I've covered most of the cases](https://github.com/Automattic/wp-calypso/pull/77667/#issuecomment-1612356342) and doesn't see any obvious breakage :)

Reply:
perfect